ModelViewer/ModelViewer/ViewerGraphicsWindow.cpp

```diff
@@ -57,6 +57,136 @@ bool ViewerGraphicsWindow::loadModel(QString filepath) {
     return m_currentModel.m_isValid;
 }
 
+bool ViewerGraphicsWindow::loadVertexShader(QString vertfilepath)
+{
+    if (!initialized)
+    {
+        return false;
+    }
+    if (vertfilepath.isEmpty()) {
+        vertfilepath = QFileDialog::getOpenFileName(nullptr, "Load Vertex Shader", "../Data/Shaders/", "");
+        if (vertfilepath.isEmpty()) {
+            return false;
+        }
+    }
+
+    //QOpenGLShaderProgram *newProgram = new QOpenGLShaderProgram(this);
+    m_program->removeAllShaders();
+
+    m_program->addShaderFromSourceFile(QOpenGLShader::Vertex, vertfilepath);
+    m_program->addShaderFromSourceFile(QOpenGLShader::Fragment, currentFragFile);
+    if (!m_program->link())
+    {
+        QString error = m_program->log();
+        qDebug() << error << endl;
+        currentVertFile = vertfilepath;
+        //m_program->addShaderFromSourceFile(QOpenGLShader::Vertex, currentVertFile);
```

Comment:
I think we delete this code since it's commented out, right?

Reply:
yeah, feel free to delete anything that's commented out